### PR TITLE
Add openapi v2 to the agent

### PR DESCRIPTION
**Pull request by knlambert**
## Description

Right now, the code base only supports Open API contract in version 3, while we want to support the version 2.

This code updates the open-api lib & use its conversion feature to migrate to version 3 when a version 2 is met.

## Testing

Tested a deployment in a test cluster with an open api v2 contract.

## Checklist

 - [x] I made sure to update `CHANGELOG.md`.

   Remember, the CHANGELOG needs to mention:
    + Any new features
    + Any changes to our included version of Envoy
    + Any non-backward-compatible changes
    + Any deprecations

 - [x] This is unlikely to impact how Ambassador performs at scale.

   Remember, things that might have an impact at scale include:
    + Any significant changes in memory use that might require adjusting the memory limits
    + Any significant changes in CPU use that might require adjusting the CPU limits
    + Anything that might change how many replicas users should use
    + Changes that impact data-plane latency/scalability

 - [ ] My change is adequately tested.

   Remember when considering testing:
    + Your change needs to be specifically covered by tests.
       + Tests need to cover all the states where your change is relevant: for example, if you add a behavior that can be enabled or disabled, you'll need tests that cover the enabled case and tests that cover the disabled case. It's not sufficient just to test with the behavior enabled.
    + You also need to make sure that the _entire area being changed_ has adequate test coverage.
       + If existing tests don't actually cover the entire area being changed, add tests.
       + This applies even for aspects of the area that you're not changing – check the test coverage, and improve it if needed!
    + We should lean on the bulk of code being covered by unit tests, but...
    + ... an end-to-end test should cover the integration points

 - [x] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.

 - [x] The changes in this PR have been reviewed for security concerns and adherence to security best practices.

### CODE REVIEW

File: pkg/agent/api_docs.go
Comment (kflynn) [inline pkg/agent/api_docs.go:262]: It bugs me that this ends up unmarshaling the V2 `docBytes` twice. Would it be cleaner to have

```
v3DocBytes, err := forceOpenAPIV3(docBytes)
```

that checks the version and upgrades if necessary? basically that's smashing `openAPIVersion` and `convertToOpenAPIV3` into a single call, since this looks like the only use for them...

**Review by kflynn — CHANGES_REQUESTED** [review]
I proposed a change to the implementation – sing out if you feel really strongly otherwise!

File: pkg/agent/api_docs.go
Comment (knlambert) [reply pkg/agent/api_docs.go]: done :)

File: docs/releaseNotes.yml
Comment (LanceEa) [inline docs/releaseNotes.yml:42]: To be clear does this make them available in the local devportal that ships with edgestack or only surfaces up in the cloud? If so it might make sense to be clear that it is the Ambassador Cloud DevPortal. If available, then ignore me :).

Comment (knlambert) [reply docs/releaseNotes.yml:42]: It will work with the local dev portal as-well, because the UI lib used support both open api 2 & 3.

The cloud version will only expose version 3 though through the conversion following one of @alexgervais comment (he was feeling like it was enough to just convert).

File: pkg/agent/api_docs.go
Comment (knlambert) [reply pkg/agent/api_docs.go:307]: Not really, and even if there was a method to get the version, it would probably unmarshal stuff in the back.
I can try the v3 first then.

File: pkg/agent/api_docs_test.go
Comment (LanceEa) [inline pkg/agent/api_docs_test.go:30]: revert per comment above

File: pkg/agent/api_docs_test.go
Comment (LanceEa) [inline pkg/agent/api_docs_test.go:98]: revert per comment above

File: pkg/agent/api_docs_test.go
Comment (LanceEa) [inline pkg/agent/api_docs_test.go:129]: revert per comment above

File: pkg/agent/api_docs_test.go
Comment (LanceEa) [inline pkg/agent/api_docs_test.go:172]: revert per comment above

File: pkg/agent/api_docs_test.go
Comment (LanceEa) [inline pkg/agent/api_docs_test.go:204]: revert per comment above

File: pkg/agent/api_docs_test.go
Comment (LanceEa) [inline pkg/agent/api_docs_test.go:348]: revert per comment above

File: pkg/agent/api_docs_test.go
Comment (LanceEa) [inline pkg/agent/api_docs_test.go:354]: revert per comment above

**Comment by LanceEa** [comment]
@kflynn - Do you have additional feedback for @knlambert on this? Would you like me to get another on the team members to give a second review?

File: pkg/agent/api_docs.go
Comment (knlambert) [reply pkg/agent/api_docs.go:307]: Yep it's basically an unmarshal. Only validate throws errors.

**Review by kflynn — DISMISSED** [review]
This looks good to me -- thanks!! 🙂

File: pkg/agent/api_docs.go
Comment (kflynn) [reply pkg/agent/api_docs.go:307]: (I had this question, too, so I was happy that the test coverage answered it for me. 🙂)

**Review by LanceEa — APPROVED** [review]
lgtm...only one nit if you want to address is the json strings in `api_docs_internal_test.go` seem to have funky spacing. Not going to hold up the PR for that.

**Review by LanceEa — APPROVED** [review]
lgtm...ci is probably a flak so I kicked it back off.

**Comment by LanceEa** [comment]
feel free to merge when ready now that CI is green!

**Pull request merged**
